src/DGtal/shapes/SurfaceMesh.h

```diff
@@ -0,0 +1,773 @@
+     be shared among faces but no correct topology is
+     required. However, you also have methods to navigate between
+     neighbor vertices, faces, etc. The mesh can be equipped with
+     normals at faces and/or vertices.
+
+     We sum up below the possible classes for representing meshes:
+     
+     - @ref Mesh represents soaps of triangles with traversal
+     operations but without any link between vertices;
+     - @ref TriangulatedSurface represents 2-manifold triangulated
+     meshes, possibly with boundaries, and provides traversal and
+     neighborhood operations (underlying fast index representation);
+     - @ref PolygonalSurface represents 2-manifold polygonal
+     meshes, possibly with boundaries, and provides traversal and
+     neighborhood operations (underlying fast index representation);
+     - @ref SurfaceMesh represents arbitrary sets of faces with
+     possible connexions between faces, edges and vertices, provides
+     traversal and neighborhood operations (underlying fast index
+     representation), and may represent non manifold meshes.
+     - @ref DigitalSurface is a common interface to represent
+     arbitrary digital surfaces, i.e. boundaries of sets of voxels
+     or interfaces between such sets;
+     - @ref IndexedDigitalSurface is a common interface to represent
+     arbitrary digital surfaces, i.e. boundaries of sets of voxels
+     or interfaces between such sets, but with underlying fast index
+     representation.
+
+     See also SurfaceMeshReader and SurfaceMeshWriter for input/output
+     operations for SurfaceMesh.
+
+     @tparam TRealPoint an arbitrary model of 3D RealPoint.
+     @tparam TRealVector an arbitrary model of 3D RealVector.
+  */
+  template < typename TRealPoint, typename TRealVector >
+  struct SurfaceMesh
+  {
+    typedef TRealPoint                              RealPoint;
+    typedef TRealVector                             RealVector;
+    typedef SurfaceMesh< RealPoint, RealVector >    Self;
+    
+    static const Dimension dimension = RealPoint::dimension;
+    BOOST_STATIC_ASSERT( ( dimension == 3 ) );
+    
+    typedef typename RealVector::Component          Scalar;
+    typedef std::vector<Scalar>                     Scalars;
+    /// The type for counting elements.
+    typedef std::size_t                             Size;
+    /// The type used for numbering vertices and faces
+    typedef std::size_t                             Index;
+    typedef Index                                   Face;
+    typedef Index                                   Edge;
+    typedef Index                                   Vertex;
+    typedef std::pair< Edge, Scalar >               WeightedEdge;
+    typedef std::pair< Face, Scalar >               WeightedFace;
+    /// The type that defines a range of vertices
+    typedef std::vector< Vertex >                   Vertices;
+    typedef std::vector< Vertex >                   VertexRange;
```

Comment:
I would only keep one of these two types

Reply:
Done